rapid7/metasploit-framework
# Use the rubyntlm gem

This changeset replaces our old vendored and modified NTLM implementation with https://github.com/WinRb/rubyntlm 
## Verification
- [x] `use auxiliary/scanner/smb/smb_login`
  - [x] **Verify** works with local accounts
  - [x] **Verify** works with AD accounts
  - [x] **Verify** works against a Domain Controller (important because of mandatory signing)
  - [ ] **Verify** bad creds don't give a false positive
- [x] `use auxiliary/scanner/mssql/mssql_login`
  - Check MSSQL authentication
    - [x] **Verify** works with MSSQL accounts (e.g. `sa`)
  - Check Windows authentication
    - [x] `set DOMAIN` to the hostname (not fqdn) of the target system. For some reason `.` and `WORKSTATION` both fail, but this is existing behavior before these changes.
    - [x] **Verify** works with local Windows accounts
    - [x] **Verify** works with AD accounts
    - [x] **Verify** bad creds don't give a false positive
- [x] `use auxiliary/scanner/mssql/mssql_hashdump` (this is important because it's different code from the login scanner)
  - Check MSSQL authentication
    - [x] **Verify** works with MSSQL accounts (e.g. `sa`)
  - Check Windows authentication
    - [x] `set DOMAIN` to the hostname (not fqdn) of the target system. For some reason `.` and `WORKSTATION` both fail, but this is existing behavior before these changes.
    - [ ] **Verify** works with local Windows accounts
    - [ ] **Verify** works with AD accounts
    - [ ] **Verify** bad creds don't give a false positive
- [x] `use auxiliary/scanner/http/http_login`
  - [x] NTLM and Negotiate are the same code, so either one will work
